config/samples/disk_formats/create-image.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+set -evx
+# Create a image with supported disk format
+#
+# The scripts assumes:
+#
+# 1. an available openstack cli
+# 2. control plane configured with disk formats
+# 2. a single layout (file / NFS) backend and disk format is deployed
+#
+#
+TIME=5
+IMAGE_NAME="myimage-disk_format-test"
+EXIT_CODE=$?
+
+
+function create_image() {
+    # This method is create, list and delete image
+    # $1 - disk format
+
+    echo This is a dodgy image > "${IMAGE_NAME}"
+
+    openstack image create \
+        --disk-format "$1" \
+        --container-format bare \
+        "${IMAGE_NAME}" 
+
+    ID=$(openstack image list | awk -v img=$IMAGE_NAME '$0 ~ img {print $2}')
+    echo "Image ID: $ID"
+    sleep "${TIME}"
+
+    if [ -z "$ID" ]
+    then
+      echo "Could not create image  " >&2
+      openstack image delete "$ID"
+      status=$(openstack image delete "$ID" | awk '/status/{print $4}')
+      printf "Image Status: %s\n" "$status"
+      exit 1
+    else
+      echo "Continue"
+    fi
+
+    # Stage 2 - Check the image is active
+    openstack image list
+    status=$(openstack image show "$ID" | awk '/status/{print $4}')
+    printf "Image Status: %s\n" "$status"
+
+    # Stage 3 - Delete the image 
+    openstack image delete "$ID"
+    status=$(openstack image delete "$ID" | awk '/status/{print $4}')
```

Comment:
Does an `image delete` status returns something like `Deleted` (I think the info are retained even if the image is no longer available) ? In this case this should also influence the exit code.
However I think you can just remove this `Stage3` (unless is useful), and create a `Stage 0` that ensures the environment is clean __before__ starting any test.

```
# Stage 0 - Delete any pre-existing image
openstack image list -c ID -f value | xargs -n 1 openstack image delete
```